src/DataExporter.ts

```diff
@@ -51,33 +57,98 @@ export class DataExporter {
     const outRow = valuesRow.join(";");
     return outRow;
   }
+  
+  private makeHeader(header: string[], statement: ECSqlStatement): string {
+    for (let i = 0; i < statement.getColumnCount(); i++) {
+      header.push(statement.getValue(i).columnInfo.getAccessString());
+    }
 
-  public writeQueryResultsToCsvFile(ecSql: string, fileName: string): void {
-    const outputFileName: string = path.join(this.outputDir, fileName);
-    const writeStream = fs.createWriteStream(outputFileName);
+    const outHeader = header.join(';');
+    return outHeader;
+  }
+  
+  private async calculateVolume(ids: Id64Array): Promise<MassPropertiesResponseProps> {
+    const requestProps: MassPropertiesRequestProps = {
+      operation: MassPropertiesOperation.AccumulateVolumes,
+      candidates: ids,
+    };
     
-    let rowCount = 0;
-    const header: string[] = [];
+    const requestContext = new BackendRequestContext();
+    const result = await this.iModelDb.getMassProperties(requestContext, requestProps);
    
-    this.iModelDb.withPreparedStatement(ecSql, (statement: ECSqlStatement): void => {
-      if (DbResult.BE_SQLITE_ROW === statement.step()) {
-        for (let i = 0; i < statement.getColumnCount(); i++) {
-          header.push(statement.getValue(i).columnInfo.getAccessString());
-        }   
-        const outHeader: string = header.join(";");
-        writeStream.write(`${outHeader}\n`);
-        writeStream.write(`${this.rowToString(statement)}\n`);
-        rowCount++;
+    return result;
+  }
+
+  public async writeQueryResultsToCsv(ecSql: string, fileName: string, options:  Partial<Options> = {}): Promise<void> {
+    const outputFileName: string = path.join(this.outputDir, fileName); 
+    
+    const opts = Object.assign({
+      calculateMassProperties: false,
+      idColumn: 0,
+      idColumnIsJsonArray: false,
+    }, options);
+
+    await this.iModelDb.withPreparedStatement(ecSql, async (statement: ECSqlStatement): Promise<void> => {     
+        if (opts.calculateMassProperties === true) {
+          await this.writeVolume(statement, outputFileName, opts);
+        } else {
+          await this.defaultWrite(statement, outputFileName);
+        }
       }
+    );
+  }
+
+  private async writeVolume(statement: ECSqlStatement, outputFileName: string, options: Options): Promise<void> {
+    const writeStream = fs.createWriteStream(outputFileName);
+    let rowCount = 0;
+    let ids: Id64Array = [];
+    const header: string[] = ["volume","area"];
+
+    let outHeader: string;
+    if (options.idColumnIsJsonArray) {
+      header.push("CodeValue");
```

Comment:
I don't understand this.  Why are you pushing CodeValue as the header? Why not just treat this like any other query and populate the headers from the statement.  

I know the id column will be nasty but that's OK ... it's not supposed to look nice it's supposed to be the raw data.